### PR TITLE
Feat: Added wishlist+refactoring

### PR DESCRIPTION
**Pull request by RahulSingh9131**
This branch contains
1. wishlist page
2. wishlist card component
3. routes for wishlist
4. context for wishlist.
5. and some refactoring

### CODE REVIEW

File: src/components/Header.js [hunk 1]
```diff
@@ -2,10 +2,12 @@ import React from 'react'
 import "../css/main.css";
 import {Link} from "react-router-dom";
 import { useCart } from '../context/CartContext';
+import { useWishlist } from '../context/WishContext';
```
Comment (rahulcs754) [inline src/components/Header.js:5]: create index file and export  from one file that will one line of code.
like 
import { useCart , useWishlist} from "../context"

File: src/components/ProductCard.js [hunk 2]
```diff
@@ -2,12 +2,13 @@ import React from 'react'
 import { ToastContainer, toast } from 'react-toastify';
 import 'react-toastify/dist/ReactToastify.css';
 import { useCart } from '../context/CartContext';
+import { useWishlist } from '../context/WishContext';
```
Comment (rahulcs754) [inline src/components/ProductCard.js:5]: same here

File: src/components/WishlistCard.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            type:"MOVE_TO_CART",
+            moveItem:{
+                _id:_id,
+                categoryName:categoryName,
+                src:src,
+                title:title,
+                originalPrice:originalPrice,
+                price:price,
+                discount:discount,
+                rating:rating
```
Comment (rahulcs754) [inline src/components/WishlistCard.js:24]: I think you can pass whole object of product here. so you have to no need to do like this

File: src/components/WishlistCard.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { useCart } from '../context/CartContext';
+import { useWishlist } from '../context/WishContext';
+
+function WishlistCard({_id,src,categoryName,price,originalPrice,title,discount,rating}){
```
Comment (rahulcs754) [inline src/components/WishlistCard.js]: set item in wishlistcart 
after function start 
do like this 
const  {_id,src,categoryName,price,originalPrice,title,discount,rating} = item ;

**Review by rahulcs754 — APPROVED** [review]
The code looks good.
just small suggestion.

File: src/components/Header.js (hunk 1 above)
Comment (RahulSingh9131) [reply src/components/Header.js:5]: ok sure , i will try

File: src/components/WishlistCard.js (hunk 3 above)
Comment (RahulSingh9131) [reply src/components/WishlistCard.js:24]: yes i tried doing that way, but somehow screen was not rendering. I think i need to look on this.

File: src/components/WishlistCard.js (hunk 4 above)
Comment (RahulSingh9131) [reply src/components/WishlistCard.js]: sure will do like this.

**Review by rohit8686 — APPROVED** [review]
I would also suggest changes that rahul has suggested, other than that it looks good to merge

**Pull request merged**